File: HelperJobby/client-app/src/DTOs/jobRelatetedDTOs/JobSalaryDTO.ts
import {ShowPayByOptions} from "../../enums/modelDataEnums/ShowPayByOptions";
import {JobDTO} from "./JobDTO";
import {SalaryRates} from "../../enums/modelDataEnums/SalaryRates";

export interface JobSalaryDTO {
    id: number;
    showPayByOption: ShowPayByOptions;
    minimalAmount: number;
    maximalAmount?: number | null;
    salaryRate: SalaryRates;
    meetsMinSalaryRequirement : boolean;
    jobId: number;
    job: JobDTO;
}

File: HelperJobby/DTOs/Job/JobSalaryDTO.cs
using ApplicationDomain.Enums;

namespace HelperJobby.DTOs.Job;

public class JobSalaryDTO
{
    public int Id { get; set; }
    public ShowPayByOptions ShowPayByOption { get; set; }
    public decimal MinimalAmount { get; set; }
    public decimal? MaximalAmount { get; set; }
    public string SalaryRate { get; set; }
    public bool MeetsMinSalaryRequirement { get; set; }
    public int JobId { get; set; }
    public JobDTO Job { get; set; }
}